csrc/transform_rfactor.cpp

```diff
@@ -116,18 +116,19 @@ class ReplayRFactor : public ReplayTransformations {
     bool static_logical_outputs = static_logical_ids_.count(s->outer()) ||
         static_logical_ids_.count(s->inner());
 
-    // Let IterDomain::split determine the correct IterType, except
-    // when the output is a reduction domain but not part of the
-    // rfactored domains. If it isn't involved in the rfactor, it's no
-    // longer a redunction domain
+    // Let IterDomain::split determine the correct IterType except when a
+    // reduction domain is split into a domain that's not involved in the
+    // rfactor. This happens when a TV is rfactored multiple times, e.g.,
```

Comment:
I'd like to have more detailed comments here, just because it took some effort to remember what's happening here. Something like:

```
Specifically, an output of a split of a reduction ID can be a non-reduction ID when the split was involved in a prior rfactor transformation. In that case, we need to preserve the non-reduction iteration type, which is not automatically done by IterDomain::split.
```